contracts/src/v0.8/dev/automation/2_1/KeeperRegistryLogicA2_1.sol

```diff
@@ -152,27 +192,70 @@ contract KeeperRegistryLogicA2_1 is
 
   function registerUpkeep(
     address target,
+    bytes4 receiver,
     uint32 gasLimit, // TODO - we may want to allow 0 for "unlimited"
     address admin,
-    bytes calldata checkData, // TODO - this should be included in the trigger
-    bytes calldata extraData
-  ) external returns (uint256 id) {
-    if (msg.sender != owner() && msg.sender != s_storage.registrar) revert OnlyCallableByOwnerOrRegistrar();
-    (Trigger triggerType, bytes memory triggerConfig, bytes memory offchainConfig) = abi.decode(
-      extraData,
-      (Trigger, bytes, bytes)
-    );
-    validateTriggerConfig(triggerType, triggerConfig);
+    bool pipelineEnabled,
+    Trigger triggerType,
+    bytes calldata pipelineData,
+    bytes memory triggerConfig,
+    bytes memory offchainConfig
```

Comment:
let's bias towards not adding complexity without proper product needs and discussions